File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        queue = deque()
        if root:
            queue.append(root)
        
        while queue:
            cur_level = []  # List to store the values of the current level
            for _ in range(len(queue)):  # Iterate over the nodes at the current level
                cur_node = queue.popleft()
                cur_level.append(cur_node.val)
                if cur_node.left:
                    queue.append(cur_node.left)
                if cur_node.right:
                    queue.append(cur_node.right)
            res.append(cur_level)
        
        return res

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> res;
        deque<TreeNode*> queue;
        if (root) queue.push_back(root);
        
        while (!queue.empty()) {
            vector<int> curLevel;  // Initialize current level vector
            int level_size = queue.size();  // Get number of nodes at current level
            
            for (int i = 0; i < level_size; i++) {
                TreeNode* curNode = queue.front();
                queue.pop_front();
                curLevel.push_back(curNode->val);
                
                if (curNode->left) queue.push_back(curNode->left);
                if (curNode->right) queue.push_back(curNode->right);
            }
            res.push_back(curLevel);
        }
        return res;
    }
};